File: apps/server/src/routes/mobileSignals.ts
// ─────────────────────────────────────────────
// apps/server/src/routes/mobileSignals.ts
//
// POST /api/mobile/signals — 接收手机信号批次
//
// TODO(phase-2): 实现 Zod 校验 MobileSignalBatch。
// TODO(phase-2): 写入 SQLite（via db/sqlite.ts）。
// TODO(phase-2): JWT 校验 userId，不信任 body 中的 userId。
// TODO(phase-2): 推送信号到 decision-engine WebSocket 通道。
// ─────────────────────────────────────────────

import { Hono } from 'hono';

export const mobileSignalsRouter = new Hono();

// TODO(phase-2): POST /signals
mobileSignalsRouter.post('/signals', async (c) => {
  // stub — 收到即返回 accepted
  return c.json({ status: 'stub', message: 'TODO(phase-2): not implemented' }, 501);
});

File: apps/server/src/api/mobile_signals.py
# TODO(phase-2): 写入 SQLite；WebSocket 推送给 decisionLoop；JWT 校验 userId。
from typing import Literal

from fastapi import APIRouter
from pydantic import BaseModel, Field

router = APIRouter(prefix="/api/mobile", tags=["mobile-signals"])


class PhoneScreenEvent(BaseModel):
    ts: int
    deviceId: str
    type: Literal["unlock", "lock", "interactive"]
    durationMs: int | None = None


class PhoneMotionEvent(BaseModel):
    ts: int
    deviceId: str
    state: Literal["stationary", "walking", "running", "vehicle", "unknown"]
    confidence: float = Field(ge=0, le=1)


class WechatStepRecord(BaseModel):
    date: str
    userId: str
    steps: int = Field(ge=0)
    source: Literal["wechat_motion", "healthkit", "google_fit"]


class ProximitySignal(BaseModel):
    ts: int
    userId: str
    source: Literal["same_wifi", "bluetooth", "local_network_ping", "manual_session"]
    nearKnownDesktop: bool
    confidence: float = Field(ge=0, le=1)
    desktopDeviceId: str | None = None


class MobileSignalBatch(BaseModel):
    userId: str
    deviceId: str
    collectedAt: int
    screenEvents: list[PhoneScreenEvent]
    motionEvents: list[PhoneMotionEvent]
    stepRecord: WechatStepRecord | None = None
    proximitySignals: list[ProximitySignal]


@router.post("/signals")
def upload_mobile_signals(batch: MobileSignalBatch) -> dict[str, int | str]:
    """Receive low-friction mobile signals.

    Persisting is intentionally left as a repository TODO. The API contract is
    already privacy-shaped: no precise location, message content, notification
    text, contact data, or raw sensor streams.
    """

    # TODO: authenticate userId from JWT instead of trusting request body.
    # TODO: write to time-series storage and expose to decisionLoop via WebSocket.
    accepted_events = (
        len(batch.screenEvents)
        + len(batch.motionEvents)
        + len(batch.proximitySignals)
        + (1 if batch.stepRecord is not None else 0)
    )
    return {"status": "accepted", "events": accepted_events}
